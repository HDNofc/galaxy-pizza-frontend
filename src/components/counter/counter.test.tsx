import userEvent from '@testing-library/user-event';
import renderWithThemeProvider from 'test-utils/render-with-theme-provider';
import Counter from './counter';

describe('<Counter>', () => {
  it('отображается корректно', () => {
    const defaultProps = {
      onMinusClick: () => {},
      onPlusClick: () => {},
    };

    const { container } = renderWithThemeProvider(<Counter {...defaultProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('отображается корректно с переданным пропсом `count`', () => {
    const defaultProps = {
      onMinusClick: () => {},
      onPlusClick: () => {},
      count: 5,
    };

    const { container } = renderWithThemeProvider(<Counter {...defaultProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('коллбеки вызываются корректно', () => {
    const onDecrement = jest.fn();
    const onIncrement = jest.fn();

    const { getByText } = renderWithThemeProvider(
      <Counter onMinusClick={onDecrement} onPlusClick={onIncrement} />
    );

    userEvent.click(getByText(/Добавить пиццу/i));
    expect(onIncrement).toHaveBeenCalled();

    userEvent.click(getByText(/Отнять пиццу/i));
    expect(onDecrement).toHaveBeenCalled();
  });
});
