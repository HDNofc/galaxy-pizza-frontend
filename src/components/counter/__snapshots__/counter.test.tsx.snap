// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter> отображается корректно 1`] = `
<div
  class="Counter-sc-9bdt3x-0 fPNYnq"
>
  <button
    class="Button-sc-1fut0te-0 DIwKh"
    type="button"
  >
    <span
      class="Text-sc-1fut0te-1"
    >
      Отнять пиццу
    </span>
    <svg>
      minus.svg
    </svg>
  </button>
  <span
    class="Value-sc-9bdt3x-1 jOvzOn"
  >
    0
  </span>
  <button
    class="Button-sc-1fut0te-0 DIwKh"
    type="button"
  >
    <span
      class="Text-sc-1fut0te-1"
    >
      Добавить пиццу
    </span>
    <svg>
      plus.svg
    </svg>
  </button>
</div>
`;

exports[`<Counter> отображается корректно с переданным пропсом \`count\` 1`] = `
<div
  class="Counter-sc-9bdt3x-0 fPNYnq"
>
  <button
    class="Button-sc-1fut0te-0 DIwKh"
    type="button"
  >
    <span
      class="Text-sc-1fut0te-1"
    >
      Отнять пиццу
    </span>
    <svg>
      minus.svg
    </svg>
  </button>
  <span
    class="Value-sc-9bdt3x-1 jOvzOn"
  >
    5
  </span>
  <button
    class="Button-sc-1fut0te-0 DIwKh"
    type="button"
  >
    <span
      class="Text-sc-1fut0te-1"
    >
      Добавить пиццу
    </span>
    <svg>
      plus.svg
    </svg>
  </button>
</div>
`;
